resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/ResourceServicePersistenceProxy.java

```diff
@@ -0,0 +1,181 @@
+     * @param request the the {@link ResourceRequest} that contains the resourceId used to retrieve resources
+     * @return a {@link Source} of {@link LeafResource}s associated with the resourceId
+     */
+    public Source<AuditableResourceResponse, NotUsed> getResourcesById(final ResourceRequest request) {
+        LOGGER.info(STORE);
+        // Try first from persistence
+        return persistence.getResourcesById(request.resourceId)
+                // If persistence returned a "cache hit"
+                .map(persisted -> persisted
+                        // Wrap with a success
+                        .map(leafResource -> AuditableResourceResponse.Builder.create()
+                                .withResourceResponse(ResourceResponse.Builder.create(request)
+                                        .withResource(leafResource)))
+                        // Persistence threw an error, create an AuditErrorMessage
+                        .recover(new PFBuilder<Throwable, AuditableResourceResponse>()
+                                .match(Exception.class, ex -> AuditableResourceResponse.Builder.create()
+                                        .withAuditErrorMessage(AuditErrorMessage.Builder.create(request,
+                                                Collections.singletonMap(ExceptionSource.ATTRIBUTE_KEY, ExceptionSource.PERSISTENCE.toString()))
+                                                .withError(new NoSuchResourceException("Exception thrown while querying persistence", ex)))).build()))
+                // If persistence is empty, a "cache miss"
+                .orElseGet(() -> Source
+                        .fromIterator(() -> {
+                            try {
+                                // Try to call out to implemented delegate service
+                                return FunctionalIterator.fromIterator(delegate.getResourcesById(request.resourceId))
+                                        // Wrap with a success
+                                        .map(leafResource -> AuditableResourceResponse.Builder.create()
+                                                .withResourceResponse(ResourceResponse.Builder.create(request)
+                                                        .withResource(leafResource)))
+                                        // Service threw an error for a single resource, create an AuditErrorMessage
+                                        .exceptionally(ex -> AuditableResourceResponse.Builder.create()
+                                                .withAuditErrorMessage(AuditErrorMessage.Builder.create(request,
+                                                        Collections.singletonMap(ExceptionSource.ATTRIBUTE_KEY, ExceptionSource.REQUEST.toString()))
+                                                        .withError(new NoSuchResourceException("Exception thrown while querying the delegate service", ex))));
+                            } catch (RuntimeException ex) {
```

Comment:
General comment we have LOGGER at the start; however we rarely log on exceptions, is this intentional? Maybe we are going to wrap the LOGGER in the exception?

Reply:
I've added a `LOGGER.error` message to the RuntimeException catch.